Content.Shared/Speech/SpeechSystem.cs

```diff
@@ -29,5 +40,44 @@ private void OnSpeakAttempt(SpeakAttemptEvent args)
             if (!TryComp(args.Uid, out SpeechComponent? speech) || !speech.Enabled)
                 args.Cancel();
         }
+
+        //#region starlight
+        public void AddMarkingEmotes(EntityUid uid, SpeechComponent component)
+        {
+            if (!TryComp<HumanoidAppearanceComponent>(uid, out var humanoidAppearance))
+                return; // no humanoid appearance so no emotes to add.
+            List<MarkingPrototype> markings = [];
+            foreach (var protoId in humanoidAppearance.MarkingSet.Markings.SelectMany(markingSet => markingSet.Value.Select(marking => marking.MarkingId)))
+            {
+                if (_prototypeManager.TryIndex<MarkingPrototype>((ProtoId<MarkingPrototype>)protoId, out MarkingPrototype? markingPrototype))
+                    markings.Add(markingPrototype!);
+            }
+            HashSet<string> AttachedIds = new();
+            foreach (var marking in markings)
+            {
+                AttachedIds.Add(marking.ID);
+            }
+
+            foreach (var marking in markings)
+            {
+                if (marking.Emotes == null)
+                    continue;
+                foreach (var emote in marking.Emotes)
+                {
+                    if (emote.RequiredMarkings == null ||
+                        AttachedIds.IsSupersetOf(emote.RequiredMarkings.Select(i => i.Id)))
+                    {
+                        Logger.Info($"Added Emote {emote.EmotePrototype.Id} to {uid}'s SpeechComponent");
```

Comment:
isnt Logger deprecated? or is that the sawmill proxy name

Reply:
print debugging. I can remove it now that it works.